File: LeetCode/784/main.py
class Solution:
    def letterCasePermutation(self, s: str) -> List[str]:
        
        def helper(s, pos) :

            if(pos == len(s)) :
                ans.append(s)
                return
            
            if(not s[pos].isalpha()) :
                helper(s, pos + 1)
            
            elif(s[pos].isupper()) :
                helper(s, pos + 1)
                temp = s[:pos] + s[pos].lower() + s[pos+1:]
                helper(temp, pos + 1)
            else :
                helper(s, pos + 1)
                temp = s[:pos] + s[pos].upper() + s[pos+1:]
                helper(temp, pos + 1)

            return

        ans = []

        helper(s, 0)

        return ans

File: LeetCode/784/main.cpp
class Solution {
public:

    vector<string> ans;

    void helper(string& str, int pos) {
        if(pos == str.length()) {
            ans.emplace_back(str);
            return;
        }

        if(!isalpha(str[pos])) {
            helper(str, pos + 1);
        } else if(islower(str[pos])) {
            helper(str, pos + 1);
            str[pos] = toupper(str[pos]);
            helper(str, pos + 1);
        } else {
            helper(str, pos + 1);
            str[pos] = tolower(str[pos]);
            helper(str, pos + 1);
        }

        return;

    } 

    vector<string> letterCasePermutation(string s) {
        helper(s, 0);
        return ans;
    }   
};